.github/workflows/preview-docs.yml

```diff
@@ -0,0 +1,97 @@
+name: Preview Documentation
+on:
+  pull_request:
+  pull_request_target:
+    types: [labeled]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  doc:
+    name: Build Documentation
+    runs-on: ubuntu-20.04
+    if: contains(github.event.pull_request.labels.*.name, 'PreviewDocumentation-accepted')
+    env:
+      PYVISTA_OFF_SCREEN: 'True'
+      ALLOW_PLOTTING: true
+      SHELLOPTS: 'errexit:pipefail'
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+          cache: 'pip'
+          cache-dependency-path: |
+            requirements.txt
+            requirements_docs.txt
+
+      - uses: awalsh128/cache-apt-pkgs-action@v1.1.3
+        with:
+          packages: libosmesa6-dev libgl1-mesa-dev python3-tk pandoc git-restore-mtime
+          version: 3.0
+
+      - name: Install PyVista and dependencies
+        run: |
+          pip install -e . --no-deps
+          pip install -r requirements_docs.txt
+
+      - name: Install custom OSMesa VTK variant
+        run: |
+          pip uninstall vtk -y
+          aria2c -x5 "https://github.com/pyvista/pyvista-wheels/raw/main/${VTK_OSMESA_WHEEL_NAME}"
+          pip install "$VTK_OSMESA_WHEEL_NAME"
+          rm "$VTK_OSMESA_WHEEL_NAME"
+        env:
+          VTK_OSMESA_WHEEL_NAME: vtk_osmesa-9.2.5-cp311-cp311-linux_x86_64.whl
+
+      - name: PyVista Report
+        run: |
+          python -c "import pyvista;print(pyvista.Report())"
+          echo PYVISTA_EXAMPLE_DATA_PATH=$(python -c "from pyvista import examples; print(examples.USER_DATA_PATH)") >> $GITHUB_ENV
+          pip list
+
+      - name: Cache Sphinx-Gallery Examples
+        uses: actions/cache@v3
+        if: env.USE_CACHE == 'true' && !startsWith(github.ref, 'refs/heads/release/') && !startsWith(github.ref, 'refs/tags/v')
+        with:
+          path: doc/source/examples/
+          key: doc-examples-${{ hashFiles('pyvista/_version.py') }}
+
+      - name: Cache example data
+        uses: actions/cache@v3
+        if: env.USE_CACHE == 'true' && !startsWith(github.ref, 'refs/heads/release/') && !startsWith(github.ref, 'refs/tags/v')
+        with:
+          path: ${{ env.PYVISTA_EXAMPLE_DATA_PATH }}
+          key: example-data-1-${{ hashFiles('pyvista/_version.py') }}
+
+      - name: Build Documentation
+        run: make -C doc html
+
+      - name: Copy ads.txt
+        run: cp doc/source/ads.txt doc/_build/html/
+
+      - name: Dump Sphinx Warnings and Errors
+        if: always()
+        run: if [ -e doc/sphinx_warnings.txt ]; then cat doc/sphinx_warnings.txt; fi
+
+      - name: Dump VTK Warnings and Errors
+        if: always()
+        run: if [ -e doc/errors.txt ]; then cat doc/errors.txt; fi
```

Comment:
If we are merging this, it would be great if it is possible to use a reusable workflow for building the documentation. This way we don't have to maintain building the documentation in both the original workflow and this new one.  Maybe a future PR...

Reply:
Very good point. I will of course need that refactoring in the future. Currently, we are trying to implement it as quickly as possible. Refactoring is not too late once this system is fully used.